cdm/core/src/test/java/ucar/ma2/TestArray.java

```diff
@@ -0,0 +1,121 @@
+    testCases.add(new Object[] {DataType.UINT, shape, new int[] {1}, new int[] {1, 1}});
+    testCases.add(new Object[] {DataType.ULONG, shape, new long[] {1}, new long[] {1, 1}});
+
+    return testCases;
+  }
+
+  private final DataType dataType;
+  private final int[] shape;
+  private final Object storage;
+  private final Object expectedCombinedArray;
+
+  public TestArray(DataType dataType, int[] shape, Object storage, Object expectedCombinedArray) {
+    this.dataType = dataType;
+    this.shape = shape;
+    this.storage = storage;
+    this.expectedCombinedArray = expectedCombinedArray;
+  }
+
+  @Test
+  public void shouldCreateArray() {
+    final Array array = Array.factory(dataType, shape);
+    assertThat(array.getShape()).isEqualTo(shape);
+  }
+
+  @Test
+  public void shouldCreateArrayWithStorage() {
+    final Array array = Array.factory(dataType, shape, storage);
+    assertThat(array.getShape()).isEqualTo(shape);
+  }
+
+  @Test
+  public void shouldCreateArrayWithIndexAndStorage() {
+    final Array array = Array.factory(dataType, Index.factory(shape), storage);
+    assertThat(array.getShape()).isEqualTo(shape);
+  }
+
+  @Test
+  public void shouldCombineTwoArrays() {
+    // tested separately below
+    assumeTrue(dataType != DataType.STRUCTURE);
+
+    final int[] combinedShape = new int[] {2};
+
+    final Array array = Array.factory(dataType, shape, storage);
+    final List<Array> arrays = new ArrayList<>();
+    arrays.add(array);
+    arrays.add(array);
+
+    final Array combinedArray = Array.factoryCopy(dataType, combinedShape, arrays);
+
+    assertThat(combinedArray.getShape()).isEqualTo(combinedShape);
+    assertThat(combinedArray.getStorage()).isEqualTo(expectedCombinedArray);
+  }
+
+  @Test
+  public void shouldCombineTwoArrayStructures() {
+    // skip structures within structures
+    assumeTrue(dataType != DataType.STRUCTURE);
+
+    final MemberBuilder memberBuilder = StructureMembers.builder().addMember("name", "desc", "units", dataType, shape);
+    final StructureMembers members = StructureMembers.builder().addMember(memberBuilder).build();
+    final StructureDataW structureData = new StructureDataW(members);
+    final Array array = Array.factory(dataType, shape, storage);
+    structureData.setMemberData("name", array);
+    final ArrayStructureW arrayStructure = new ArrayStructureW(structureData);
+
+    final int[] combinedShape = new int[] {2};
+    final List<Array> arrays = new ArrayList<>();
+    arrays.add(arrayStructure);
+    arrays.add(arrayStructure);
+
+    final ArrayStructure combinedArray = (ArrayStructure) Array.factoryCopy(DataType.STRUCTURE, combinedShape, arrays);
+
+    assertThat(combinedArray.getShape()).isEqualTo(combinedShape);
+    assertThat(combinedArray.getArray(0, members.getMember(0))).isEqualTo(array);
+    assertThat(combinedArray.getArray(1, members.getMember(0))).isEqualTo(array);
```

Comment:
Will this work if the `ArrayStructures` don't have the same members?

Reply:
Yeah it probably would fail but now I added a check with an exception for this situation and added a couple more tests for the `ArrayStructures`. In order to check the `StructureMembers` are the same I also added an `StructureMembers::equals` and tests for that.